can/message.py

```diff
@@ -65,7 +65,7 @@ def __str__(self):
 
         field_strings.append("DLC: {0:d}".format(self.dlc))
         data_strings = []
-        if self.data is not None:
+        if self.data is not None and len(self.data) > 0:
             for index in range(0, self.dlc):
```

Comment:
Isn't the core issue that we are iterating to `dlc` instead of iterating over the data array?